### ISSUE TITLE
DraftProfile is out of sync with Profile domain object

### ISSUE THREAD

**Issue by temi** [AtlasOfLivingAustralia/profile-hub#543]
An example of this is machedName property. In Profile domain object it is an embedded property but in DraftProfile it is not. Because it is not embedded in DraftProfile it has caused mongo to create a collection called Name. Please list other instances below.

**Comment by ben3000** [comment]
#414

**Comment by ben3000** [comment]
DraftProfile is a Mongo embedded sub-document and thus its properties are managed manually throughout profile-hub, causing a range of issues in the application. A draft profile could instead be a reference from a Profile to another Profile noting which is parent and which is child. This would avoid the problem of diverging properties in DraftProfile and Profile.

**Comment by temi** [comment]
I agree.. but need to do some testing on what the consequences would be.

**Comment by sbearcsiro** [comment]
Another option would be to extract the common properties into a mixin-able or inheritable type, which would theoretically mean fewer changes to the existing code that deals with drafts and as a bonus make a theoretical chain of drafts (Profile(A).draft -> Profile(B).draft -> Profile(C).draft) impossible

If the @delegate annotation works in GORM domain objects then that might also allow us to just compose the core attributes instead of using inheritance.

**Comment by ben3000** [comment]
Apparently [@Delegate doesn't actually work in domain objects](https://stackoverflow.com/questions/40373021/grails-3-domain-composition-using-delegate), so perhaps the mixin is the only other option.
